packages/slate-plugins/src/elements/mention/transforms/wrapMentionBrackets.ts

```diff
@@ -0,0 +1,63 @@
+import { Point, Range, Editor, Transforms } from "slate";
+
+const pointToStr = (point: Point | undefined) => {
+  if (!point) {
+    return "-";
+  }
+  let pointStr = "";
+  pointStr += point.path.join(".");
+  pointStr += " " + point.offset;
+  return pointStr;
+};
+
+const rangeToStr = (range: Range) => {
+  let rangeStr = "";
+  rangeStr += pointToStr(range.anchor);
+  rangeStr += " -> ";
+  rangeStr += pointToStr(range.focus);
+  return rangeStr;
+};
+
+export const wrapMentionBrackets = (editor: Editor, selection: Range) => {
+  console.dir(rangeToStr(selection));
+
+  const containingNode = Editor.node(editor, selection);
+  console.log("containing node:");
+  console.dir(containingNode);
+
+  const marksObj = Editor.marks(editor);
+  const marks = marksObj ? Object.keys(marksObj) : [];
+
+  if (!containingNode[0].text || marks.length > 0) {
+    console.log("cannot wrap a selection that is not simple text, ignoring");
+    return;
+  }
+
+  const anchor = selection.anchor;
+  const focus = selection.focus;
+
+  const leftPoint = Range.isBackward(selection) ? focus : anchor;
+  const rightPoint = Range.isBackward(selection) ? anchor : focus;
+
+  Transforms.insertText(editor, "[", { at: leftPoint });
+  const newRightPoint =
+    Editor.after(editor, rightPoint, {
+      distance: 1,
+      unit: "character",
+    }) || rightPoint;
+
+  Transforms.insertText(editor, "]", { at: newRightPoint });
+  const newLeftPoint =
+    Editor.after(editor, leftPoint, {
+      distance: 1,
+      unit: "character",
+    }) || leftPoint;
+
+  const newAnchor = Range.isBackward(selection) ? newRightPoint : newLeftPoint;
+  const newFocus = Range.isBackward(selection) ? newLeftPoint : newRightPoint;
```

Comment:
same as above, replaceable by https://docs.slatejs.org/api/locations#range-edges-range-range-options-point-point

Reply:
fixed

Reply:
wait - can you explain how? this is the reverse operation

Reply:
`Range.edges(selection, {selection: Range.isB`